content/news/posts/2024/meet-prestashop9-api.md

```diff
@@ -0,0 +1,386 @@
+
+![PrestaShop 9 API Client](/assets/images/2024/05/api/api_add_new_client1.jpeg)
+
+If you want to add a new client, you must provide information about it. In the form below, you must provide all the necessary information about the client. After that, you can click the "Generate client secret and save" button. This will generate a client secret for you and save the client in the database. **It's important to save the client secret because you won't be able to see it again.**
+
+![PrestaShop 9 Form for adding a new API client](/assets/images/2024/05/api/api_add_new_client2.jpeg)
+<p class="text-center text-muted small">Form for adding a new API client</p>
+
+* **Client name** - the name of the client: it can be anything you want. It's just for your reference.
+* **Client ID** - the ID of the client, which is a unique identifier for the client. It should be written without spaces or special characters.
+* **Description** - a description of the client: it can be anything you want. It's just for your reference.
+* **Lifetime** - the lifetime of the client. It's the time after which the client access token will expire. In seconds.
+* **Enabled** - if the client is enabled or not. If it's not enabled, the client won't be able to use the API.
+* **Scopes** - the scopes the client can use. You can enable multiple scopes. Each time you add a new scope, you will have to adjust the client's settings.
+
+{{% notice %}}
+As part of API extensibility, you can add new scopes to the API. This will allow you to create more granular permissions for your clients. You can find an [example module here](https://github.com/PrestaShop/example-modules/tree/master/api_module) and in [ps_apiresources](https://github.com/PrestaShop/ps_apiresources) repository.
+{{% /notice %}}
+
+After you save the client, you will see the client secret, but you won't be able to see it again. You can only see it once when you generate it. It's important to save it because you need it to authenticate your client.
+
+![PrestaShop 9 API client secret](/assets/images/2024/05/api/api_add_new_client3.jpeg)
+<p class="text-center text-muted small">Remember to copy secret for your API Client</p>
+
+## Client credentials grant
+
+Now that you created your first API client, you can test the connection to the API. You can send a GET request to the `/admin-api` endpoint. You need to provide the client ID and client secret you generated earlier. You can use Postman or any other tool to make HTTP requests.
+
+{{% notice %}}
+You have many tools to debug APIs. You can use Postman, [curl](https://curl.se/), or any other tool (even open source like [Insomnia](https://insomnia.rest/)) that allows you to make HTTP requests. In this tutorial, I will use [Postman API client](https://www.postman.com/api-platform/api-client/), so that you can quickly test the API with a user-friendly interface.
```

Comment:
Insomnia is freemium now, like POstman But you have Insomnium, although the repository seems to have been archived I don't know why 🤔